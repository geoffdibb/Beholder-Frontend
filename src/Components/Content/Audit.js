import React from 'react';
import { Button, Form, Col, Row, FormGroup, Label, Input, Table } from 'reactstrap';
import axios from "axios";

export default class Audit extends React.Component {

     constructor() {
        super()
        this.state = {
            responseauditrequestlogData: [],
            responsesearchlogData: [],
            responseaudituseraccesslogData: []
        };

    }

    //auditrequests
    searchauditrequestlog = (e) => {
        e.preventDefault();
        const headers = {
            'Content-Type': 'application/json',
            'Authorization': this.props.apitoken
        }
        axios
            .get("http://localhost:5001/getauditrequestlog/" + this.props.username, { headers })
            .then(response => {
                this.setState({
                    responseauditrequestlogData: (response.data)
                });
            })
            .catch(err => { this.setState({ message: "Audit logs not found" }); });

    }

    
    //searchlogs
    searchauditsearchlogs = (e) => {
        e.preventDefault();
        const headers = {
            'Content-Type': 'application/json',
            'Authorization': this.props.apitoken
        }
        axios
            .get("http://localhost:5001/getsearchlog/" + this.props.username, { headers })
            .then(response => {
                this.setState({
                    responsesearchlogData: (response.data)
                });

                console.log(this.state.responsesearchlogData)
            })
            .catch(err => {
                this.setState({ message: "Search logs not found" });
            });

    }


    //accesslogs
    searchaudituseraccesslog = (e) => {
        e.preventDefault();
        const headers = {
            'Content-Type': 'application/json',
            'Authorization': this.props.apitoken
        }
        axios
            .get("http://localhost:5001/getaudituseraccesslog/" + this.props.username, { headers })
            .then(response => {
                this.setState({
                    responseaudituseraccesslogData: (response.data)
                });
            })
            .catch(err => {
                this.setState({ message: "User logs not found" });
            });
    }

    renderLog(response, index) {
        return (
            <tr key={index}>
                <td>{response.username}</td>
            </tr>
        )
    }

    render() {
        return (
            <div>
                <Col md='2'></Col>
                <div className="p-2 bg-info my-1 rounded">
                    <Row>

                        <Row style={{ justifyContent: 'center', alignItems: 'center' }}>
                            <Col sm={{ size: '5', offset: 1 }}>
                                <p>{this.state.responseauditrequestlogData}</p>
                                <button onClick={this.searchauditrequestlog}> Retrieve audit request Logs </button>

                            </Col>
                            <Col>
                                <p>{this.state.searchauditsearchlogs}</p>
                                <button onClick={this.searchauditsearchlogs}> Retrieve search Logs </button>
                            </Col>
                            <Col>
                                <p>{this.state.responseaudituseraccesslogData}</p>
                                <button onClick={this.searchaudituseraccesslog}> Retrieve user access Logs </button>
                            </Col>
                        </Row>
                    </Row>
                    <Row>
                        <p>{this.state.responseauditrequestlogData}</p>
                        <p>{this.state.searchauditsearchlogs}</p>
                        <p>{this.state.responseaudituseraccesslogData}</p>
                        <Table striped condensed hover>
                            <thead>
                                <tr>
                                    <th></th>
                                    <th>Name</th>
                                    <th>Address</th>
                                    <th>Age</th>
                                </tr>
                            </thead>
                            <tbody>
                                {this.state.responsesearchlogData.map(this.renderLog)}
                            </tbody>
                        </Table>

                    </Row>

                </div>

            </div>
        );
    };

};