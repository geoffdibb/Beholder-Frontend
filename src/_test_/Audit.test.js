import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import Audit from '../Components/Content/Audit';

import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer';


Enzyme.configure({ adapter: new Adapter() })

it('snapshot Audit', () => {
  const tree = renderer.create(<Audit/>).toJSON()
  expect(tree).toMatchSnapshot();
})