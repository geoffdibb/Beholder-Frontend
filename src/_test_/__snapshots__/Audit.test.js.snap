// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Audit 1`] = `
<div
  className="p-2 bg-info my-1 rounded"
>
  <br />
  <div
    className="row"
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <div
      className="col-sm-3 offset-sm-0"
    >
      <button
        aria-label={null}
        className="btn btn-secondary"
        onClick={[Function]}
        type="button"
      >
         Retrieve audit request Logs 
      </button>
    </div>
    <div
      className="col-sm-3 offset-sm-1"
    >
      <button
        aria-label={null}
        className="btn btn-secondary"
        onClick={[Function]}
        type="button"
      >
         Retrieve search Logs 
      </button>
    </div>
    <div
      className="col-sm-3 offset-sm-1"
    >
      <button
        aria-label={null}
        className="btn btn-secondary"
        onClick={[Function]}
        type="button"
      >
         Retrieve user access Logs 
      </button>
    </div>
  </div>
  <div
    className="row"
  >
    <table
      className="table"
    />
  </div>
</div>
`;
